prep.sh

```diff
@@ -80,17 +91,15 @@ function setup_systemd() {
     # Network configuration
     cp /config/*.network  ${MOUNTROOT}/etc/systemd/network
     echo "Reloading the config files."
-    systemctl daemon-reload
+    systemd_daemon_reload
     # Make sure we're actually waiting for the network if it's required.
-    systemctl --root=${MOUNTROOT} enable systemd-networkd-wait-online.service
+    systemd_enable_units systemd-networkd-wait-online.service
     echo "ENABLing all config files."
-    find ${MOUNTROOT}/etc/systemd/system -maxdepth 1 ! -name "*.sh" -type f | \
-        xargs --no-run-if-empty basename -a | \
-        xargs --no-run-if-empty systemctl --root=${MOUNTROOT} enable
+    systemd_enable_units $(find ${MOUNTROOT}/etc/systemd/system -maxdepth 1 ! -name "*.sh" -type f | \
+      xargs --no-run-if-empty basename -a )
     echo "RESTARTing all .path files."
-    find ${MOUNTROOT}/etc/systemd/system -maxdepth 1 -name "*.path" -type f | \
-        xargs --no-run-if-empty basename -a | \
-        xargs --no-run-if-empty systemctl restart
+    systemd_enable_units $(find ${MOUNTROOT}/etc/systemd/system -maxdepth 1 -name "*.path" -type f | \
```

Comment:
Will run with empty list because of `if [ $# -eq 0 ]; then return; fi`
As for the length, there is not safety measure, but the actual length will be orders of magnitude smaller than max.